### PR TITLE
Unused parameter elimination

### PR DESCRIPTION
**Pull request by jspam**

### CODE REVIEW

File: src/main/scala/mjis/opt/UnusedParameterElimination.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+object UnusedParameterElimination extends Optimization {
+  override def optimize(): Boolean = {
+    val callers = CallGraph.callerMap()
+    CallGraph.graphsInTopologicalOrder.map(
+      g => if (g.methodType.getNParams > 0) _optimize(g, callers(g)) else false
+    ).toList.contains(true)
+  }
+
+  protected def _optimize(g: Graph, callers: Seq[Call]): Boolean = {
+    BackEdges.enable(g)
```
Comment (TehMillhouse) [inline src/main/scala/mjis/opt/UnusedParameterElimination.scala]: Couldn't you just make this an `Optimization(needsBackEdges = true)` or is there a readon not to?

**Comment by TehMillhouse** [comment]
Other than that one comment, LGTM

File: src/main/scala/mjis/opt/UnusedParameterElimination.scala (hunk 1 above)
Comment (jspam) [inline src/main/scala/mjis/opt/UnusedParameterElimination.scala]: The reason is I need to pass the `callers` parameter to `_optimize`. Do you see a better way of doing that?

Comment (TehMillhouse) [inline src/main/scala/mjis/opt/UnusedParameterElimination.scala]: Oh, I hadn't realized you need to override `optimize()` to ensure the right optimization ordering. Never mind then.

**Pull request merged**